jib-core/src/main/java/com/google/cloud/tools/jib/image/LayerEntry.java

```diff
@@ -16,37 +16,92 @@
 
 package com.google.cloud.tools.jib.image;
 
-import com.google.common.collect.ImmutableList;
+import com.google.common.annotations.VisibleForTesting;
 import java.nio.file.Path;
+import java.util.Objects;
 
 /**
  * Represents an entry in the layer. A layer consists of many entries that can be converted into tar
  * archive entries.
  */
 public class LayerEntry {
 
+  private final Path sourceFile;
+  private final Path extractionPath;
+
   /**
-   * The source files to build from. Source files that are directories will have all subfiles in the
-   * directory added (but not the directory itself).
+   * Instantiates with a source file and the path to place the source file in the container file
+   * system.
```

Comment:
It'd be nice to eliminate any ambiguity with an example. I believe it is not like `extractionPath` is a directory and the file will be placed inside `extractionPath`, put rather `extractionPath` is the final path of the file, right?

Reply:
Yes, it is the final path. Will add an example.